spec/views/hq/users/index_spec.rb

```diff
@@ -2,51 +2,81 @@
+      end
+
+      it 'hides the table' do
+        expect(rendered).to match /No Users Found/
+        expect(rendered).to have_selector('table#users-index', count: 0)
+      end
+
+      it 'hides the pagination bar' do
+        expect(rendered).to_not have_selector('.pagination', text: /Previous(.*)1(.*)Next/)
       end
     end
 
     context '1 user' do
-      it 'displays a single user' do
-        @users = [ double('User', user_name: 'Frederick Bloggs', 
-          email: 'fred.bloggs@example.com', active_sponsors: (1..5).to_a)]
+      before(:each) do
+        @users = [ FactoryGirl.build_stubbed(:user, 
+                   user_name: 'Frederick Bloggs',
+                   email: 'fred.bloggs@example.com') ].paginate
         render
-        expect(rendered).to have_selector('table tr', count: 2)
-        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
+      end
+
+      it 'displays a single user' do
+        expect(rendered).to have_selector('table tbody tr', count: 1)
+        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users.first))
         expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
+      end
+
+      it 'hides the pagination bar' do
+        expect(rendered).not_to have_selector('.pagination', text: /Previous(.*)1(.*)Next/)
       end
     end
 
-    context '>1 user' do
+    context 'many users on multiple pages' do
+      before :each do
+        @users= [ FactoryGirl.build_stubbed(:user),
+                  FactoryGirl.build_stubbed(:user, user_name: 'Joseph Soap',
+                    email: 'joe.soap@example.com'),
+                  FactoryGirl.build_stubbed(:user, user_name: 'Tom Jones',
+                    email: 'tom.jones@example.com') ].paginate(:page => 2, :per_page => 2)
+      end
+
       it 'displays multiple users' do
-        @users = [ double('User', user_name: 'Frederick Bloggs', 
-          email: 'fred.bloggs@example.com', active_sponsors: (1..5).to_a), 
-                   double('User', user_name: 'Joseph Soap', 
-          email: 'joe.soap@example.com', active_sponsors: (1..7).to_a) ]
         render
-        expect(rendered).to have_selector('table tr', count: 3)
-        expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
-        expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
-        expect(rendered).to have_link('Joseph Soap', href: hq_user_path(@users[1]))
-        expect(rendered).to match 'joe.soap@example.com'
-        expect(rendered).to match '7'
+
+        expect(rendered).to have_selector('table tbody tr', count: 1)
+        expect(rendered).to have_link('Tom Jones', href: hq_user_path(@users.first))
+        expect(rendered).to match 'tom.jones@example.com'
+        expect(rendered).to_not have_link('Joseph Soap')
+        expect(rendered).to_not match 'joe.soap@example.com'
```

Comment:
To answer questions like that for myself, I ask, "Whose code is making that decision?" If it's ours, test it. If it's from a gem, don't.